src/prometheus_alert_rules/loki_disk_will_be_filled.rule

```diff
@@ -0,0 +1,8 @@
+alert: LokiDiskIsFilling
+expr: ((sum(node_filesystem_free_bytes{mountpoint="/loki/chunks"}) / sum(rate(loki_ingester_chunk_stored_bytes_total[5m]))) / 360) < 24
```

Comment:
- Should we add a comment explaining what's going on in this expr? E.g. `# Current hourly ingestion rate will fill up disk in less than 24 hours`
- Iirc, disk usage alerts are typically phrased with `predict_linear`.
- I know the `_total` suffix means it's a counter, but it's odd to use `rate` instead of `deriv`. We have compaction cycles and cleanup cycles that would not be accounted. Perhaps there's a better metric? Or, perhaps we could rely only on `predict_liner` of `node_filesystem_free_bytes` alone?

Reply:
> * Should we add a comment explaining what's going on in this expr? E.g. `# Current hourly ingestion rate will fill up disk in less than 24 hours`

Yes, make sense!

 
> * Iirc, disk usage alerts are typically phrased with `predict_linear`.

What do you think about this:

```promql
predict_linear(node_filesystem_free_bytes{mountpoint="/loki/chunks"}[5m], 60*60*24) < 0
```

Seems even an easier query...

 
>  * I know the `_total` suffix means it's a counter, but it's odd to use `rate` instead of `deriv`. We have compaction cycles and cleanup cycles that would not be accounted. Perhaps there's a better metric? Or, perhaps we could rely only on `predict_liner` of `node_filesystem_free_bytes` alone?

AFAIU there is no better metric than this one... anyway, if we use the query above it's ok.

Reply:
- `mountpoint` is added automatically by node-exporter:

```
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/charm/bin/pebble"} 1.311690752e+09
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/charm/container"} 1.311690752e+09
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/etc/hostname"} 1.311690752e+09
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/etc/hosts"} 1.311690752e+09
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/etc/resolv.conf"} 1.311690752e+09
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/loki/boltdb-shipper-active"} 1.311690752e+09
node_filesystem_free_bytes{device="/dev/sda1",fstype="ext4",mountpoint="/loki/chunks"} 1.311690752e+09
node_filesystem_free_bytes{device="tmpfs",fstype="tmpfs",mountpoint="/var/run/secrets/kubernetes.io/serviceaccount"} 8.20953088e+09
```

We add the topology on top of that.
Since we may have many devices, that's why in the example I hardcoded the `mountpoint`. We want to know when the `device` in which `/loki/chunks` is mounted is going to be filled. As you pointed out it's risky, we can forget to change it if we change `metadata.yaml`

- About [rule unit test](https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/)... we can add it... in fact seems interesting!

- `node_filesystem_free_bytes` is "Filesystem free space in bytes." whereas `node_filesystem_avail_bytes` is "Filesystem space available to non-root users in bytes." ... from my point of view it doesn't matter the available space to non-root users